### ISSUE TITLE
Training and inference hyperparameters to reproduce results

### ISSUE THREAD

**Issue by shirley-wu** [allenai/openpi-dataset#13]
Hi, I am trying to reproduce your results reported in the paper. I use the default training and inference hyperparameters but get the following results (much lower than in your paper). Could you release the detailed hyperparameters? Thanks!

| | F1 (exact) | F1 (bleu) | F1 (rouge) |
|---|---|---|---|
| Reported in paper | 4.3 | 16.1 | 32.4 |
| Mine | 2.35 | 11.94 | 27.35 |

My settings:
* Training: I use default hyperparameters in `scripts/training_bash.sh` on 1 GPU
* Generation: I use `training/generation.py` with `--max_len 200`. Other hyperparameters are as default. Input file is `data/formatted_for_gpt2/test.jsonl`
* Evaluation: I evaluate formatted output against `data/gold-v1.1/test.jsonl` (I also tried `data/gold/test/id_answers.jsonl` but the results are almost the same)

**Comment by ToneLi** [comment]
sorry to bother you, have you solved the problem?  and got the detailed hyperparameters? I struggled with it for many weeks. Thanks!!

**Comment by nikett** [comment]
Hi Tone Li,

I am very sorry for the late reply. I am attending to this today.

I received three emails at exactly the same time and I solved the problem
for one of them. I didn't realize that all three were from different
institutes.

**Comment by ToneLi** [comment]
you can list the hyperparm in Github (the command), I send the email (gsu.edu) last day.  I sincerely hope you can tell me the hyperparamers today

**Comment by ToneLi** [comment]
I think you can tell us the hyperparamers you used, and how can I get the results reported in the paper, I think it is helpful for our research

**Comment by ToneLi** [comment]
I am waiting for your comments, thanks !! thanks again!!

**Comment by nikett** [comment]
Please use the following hyperparameters.

https://github.com/allenai/openpi-dataset/blob/main/hyperparams.md

**Comment by ToneLi** [comment]
thanks lot

**Comment by ToneLi** [comment]
do we have this file? run_trackworld.py

**Comment by ToneLi** [comment]
![image](https://user-images.githubusercontent.com/33856565/156830829-d5e3f8bb-8ff1-4904-8ba5-4564a83ace21.png)

**Comment by nikett** [comment]
It should be `run_trainer.py` as mentioned in the training bash.

**Comment by ToneLi** [comment]
Got it

**Comment by ToneLi** [comment]
Another question is in the run_trainer.py, there is no parameter "window_size"
But in the command, you given the value about window_size=1, can I remove it? what should I do?
![1646427367(1)](https://user-images.githubusercontent.com/33856565/156840321-c2133e90-38e3-4c9a-97fe-ce033365da5e.png)

**Comment by nikett** [comment]
yes please ignore it.

**Comment by ToneLi** [comment]
thanks lot!!

**Comment by shirley-wu** [comment]
Hi @ToneLi , sorry for the late response.

I emailed Niket previously and he sent me his script (the one he posted here). I followed the script and achieve results near in the paper. After fixing some issues (see below), my final score so far is F exact / rouge / bleu = 4.23 / 18.21 / 35.66 (even higher than the paper)

Here are my training script. I think the main trick is using GPT2-medium:
```bash
MODEL_TYPE=${1:-"gpt2"}
MODEL_NAME=${2:-"gpt2-medium"}

WEIGHT_DECAY=${3:-"0.0"}
WARMUP=${4:-"0"}
LR=${5:-"5e-05"}
ACCUM=${6:-"2"}
NUM_EPOCHS=${7:-"30"}

MAX_LEN=20
BATCH_SIZE_TRAIN=4  # cannot be any larger for my GPU. Almost reach OOM
BATCH_SIZE_EVAL=16
BLOCK_SIZE=300

EXP=exps/${MODEL_NAME}

PYTHONPATH=$PYTHONPATH:. python training/run_trainer.py \
    --output_dir=$EXP \
    --warmup_steps $WARMUP \
    --model_type="$MODEL_TYPE" \
    --model_name_or_path=${MODEL_NAME} \
    --do_train \
    --train_data_file=data/formatted_for_gpt2/train.jsonl \
    --per_gpu_train_batch_size $BATCH_SIZE_TRAIN \
    --per_gpu_eval_batch_size $BATCH_SIZE_EVAL \
    --gradient_accumulation_steps $ACCUM \
    --overwrite_output_dir \
    --length $MAX_LEN \
    --block_size $BLOCK_SIZE \
    --save_total_limit 3 \
    --save_steps 1000 \
    --learning_rate $LR \
    --weight_decay $WEIGHT_DECAY \
    --num_train_epochs $NUM_EPOCHS \
    --overridden_model_configs '{"resid_pdrop": 0.1, "attn_dropout": 0.1}'
```

For evaluation, I use max length = 400 instead of 100

A few issues in evaluation and data:
1. Deduplication issue and split issue: see https://github.com/allenai/openpi-dataset/pull/14. You may want to pull this repo again
2. In GPT-2 training data, some sentences are represented as "xx of xx **were** xx before and xx afterwards" (using **were** instead of **was**). But the evaluation script filters out sentences using "were". I got a higher score by replacing all "were" with "was" in the output. I think this is a training-evaluation mismatch. I've sent more details to Niket and I think he's looking into it.

**Comment by ToneLi** [comment]
ok thanks for your reply,  your information is very useful

**Comment by ToneLi** [comment]
another thing, I found the author used question and answers are the input, is it reasonable? why do not choose the question as  the model input? 
![image](https://user-images.githubusercontent.com/33856565/156927551-fcdeb59e-4042-48d3-82f9-5176f62f3474.png)

**Comment by shirley-wu** [comment]
> another thing, I found the author used question and answers are the input, is it reasonable? why do not choose the question as the model input? ![image](https://user-images.githubusercontent.com/33856565/156927551-fcdeb59e-4042-48d3-82f9-5176f62f3474.png)

Yes I think it's reasonable. This is because GPT-2 model is an autoregressive model so input_ids contain both input question and output answer. If you look into the labels you'll see the loss is only calculated on the answer.
